### ISSUE TITLE
Issue with expandable connectors in Buildings 7.0.1

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#7833]
Please check 

[Buildings.Applications.DataCenters.ChillerCooled.Examples.IntegratedPrimaryLoadSideEconomizer](https://libraries.openmodelica.org/branches/master/Buildings_7.0.0/files/Buildings_7.0.0_Buildings.Applications.DataCenters.ChillerCooled.Examples.IntegratedPrimaryLoadSideEconomizer.err
)

The NF complains that
```
[Buildings 7.0.1/Applications/DataCenters/ChillerCooled/Examples/BaseClasses/PartialDataCenter.mo:344:3-351:29:writable]
Error: Cannot connect expandable connector weaData.weaBus with non-expandable connector weaBus.TWetBul.
```
@perost, is this an NF bug or are there problems with the validity of the source code?

**Comment by casella** [comment]
@perost, this is the last pending NF issue of the Buildings library, with three failing models.

Very good job so far!

**Comment by casella** [comment]
Possibly addressed by @mwetter in Buildings [ticket #2706](https://github.com/lbl-srg/modelica-buildings/issues/2706)

**Comment by perost** [comment]
Fixed in the library, and the affected models now simulate successfully.

The actual issue was that we don't support adding new expandable connectors via connections, e.g. `connect(ec.ec2.x, x)` where `ec` is an expandable connector and `ec2` is undeclared. Since this only occurred in the library by mistake I won't open a new ticket for it, since it likely won't be implemented until someone asks for it anyway.
